kinesis/kinesis.go

```diff
@@ -562,15 +562,32 @@ func (outputPlugin *OutputPlugin) randomString() string {
 	return string(outputPlugin.random.buffer)
 }
 
+func getFromMap(dataKey string, records map[interface{}]interface{}) interface{} {
+	for k, v := range records {
+		currentKey := stringOrByteArray(k)
+		if currentKey == dataKey {
+			return v
+		}
+	}
+
+	return nil
+}
+
 // getPartitionKey returns the value for a given valid key
 // if the given key is empty or invalid, it returns a random string
 func (outputPlugin *OutputPlugin) getPartitionKey(record map[interface{}]interface{}) string {
 	partitionKey := outputPlugin.partitionKey
+	partitionKeys := strings.Split(partitionKey, "->")
+	num := len(partitionKeys)
 	if partitionKey != "" {
```

Comment:
the check for `if partitionKey != "" {` should be first, right?

Reply:
Make sense